youtube_dl/extractor/visir.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    base_url,
+    remove_start,
+    urljoin,
+)
+
+
+class VisirMediaIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?visir\.is/section(?:/media)?/.+?fileid=(?P<id>[^/]+)$'
+    _TESTS = [{
+        'url': 'http://www.visir.is/section/MEDIA99&fileid=CLP51729',
+        'md5': '1486324696d1b9f30fcea985a7922f2c',
+        'info_dict': {
+            'id': 'CLP51729',
+            'display_id': 'CLP51729',
+            'ext': 'mp4',
+            'title': 'Gu\u00f0j\u00f3n: Mj\u00f6g j\u00e1kv\u00e6\u00f0ur \u00e1 framhaldi\u00f0',
+            'description': None,
+            'thumbnail': 'http://www.visir.is/apps/pbcsi.dll/urlget?url=/clips/51729_3.jpg'
+        },
+    }, {
+        'url': 'http://www.visir.is/section/MEDIA99&fileid=CLP45905',
+        'info_dict': {
+            'id': 'CLP45905',
+            'display_id': 'CLP45905',
+            'ext': 'mp4',
+            'title': 'Eva Laufey - Nau\u00f0synlegt a\u00f0 b\u00f6rn f\u00e1i a\u00f0 koma n\u00e1l\u00e6gt matarger\u00f0',
+            'description': 'md5:24422433a08d270a3690d149edf113b8',
+            'thumbnail': 'http://www.visir.is/apps/pbcsi.dll/urlget?url=/clips/45905_3.jpg',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    @staticmethod
+    def _extract_urls(webpage):
+        media_base_url = 'http://www.visir.is/section/media/?template=iplayer&fileid=%s'
+        video_ids = [media_base_url % m.group('id') for m in re.finditer(
+            r'App\.Player\.Init\(\{[^\}]*Type:\s*\'(?:audio|video)\'[^\}]+FileId:\s*\'(?P<id>.+?)\'[^\}]+Host:\s*\'visirvod\.365cdn\.is\'',
+            webpage)]
+        return video_ids
+
+    def _extract_formats(self, filename, video_id, media_type):
+        playlist_url = 'http://visirvod.365cdn.is/hls-vod/_definst_/mp4:%s/playlist.m3u8' % filename
```

Comment:
@dstftw Do you know, if there is a similar json file for audio shows? Because e.g. for http://www.visir.is/section/MEDIA98&fileid=CLP49923, the url http://www.visir.is/section/MEDIA?template=related_json&kat=10&subkat=14 does not work...